### PR TITLE
solution 2.3

### PR DESCRIPTION
**Pull request by K-ratos**

### CODE REVIEW

File: solutions/abhishek-kujur/aserver.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+
+class Server():
+    def __init__(self, host, port):
+        self.host = host
+        self.port = port
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.socket.bind((self.host, self.port))
+        self.logger = logging.getLogger('Server')
+        log_filename = 'Server.log'
```
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: This should be configurable. In general, it is reasonable to take a logger as input, and allow the provider to add handlers that can print to stdout/write to file, etc.

File: solutions/abhishek-kujur/aserver.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            return
+        if(path == '/'):
+            path = ''
+        file_to_check = '.' + path.replace("%20", " ")
+        self.logger.info(file_to_check)
+        self.logger.info("is a file" + str(os.path.isfile(file_to_check)))
+        if os.path.isdir(file_to_check):
+            directory_list = os.listdir(path=file_to_check)
+            if 'index.html' in directory_list:
+                html = open(file_to_check+'/index.html', 'r')
```
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: Use `os.path.join`

File: solutions/abhishek-kujur/aserver.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                r = html.read()
+            else:
+                r = "<html>\n<body>\n"
+                for each in directory_list:
+                    r += "<a href=" + "'" + path + "/" + each + "'" + ">" + each + "</a><br>\n"
```
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: Use <ul> / <li>. No need for \n here.

File: solutions/abhishek-kujur/aserver.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                r += "</body>\n</html>"
+            connection.send('HTTP/1.0 200 OK\r\n'.encode())
+            connection.send("Content-Type: text/html\r\n".encode())
```
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: You should make this configurable. Maintain a map<header-name, value> that you can loop over to generate the headers.

File: solutions/abhishek-kujur/aserver.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        elif os.path.isfile(file_to_check):
+            file_type = os.path.splitext(path)[-1].lower()
+            file_type = file_type[1:len(file_type)]
+            size_of_file = os.path.getsize(file_to_check)
+            header_content_type = "Content-Type: " + file_type + "\r\n"
+            header_content_length = "Content-Length: " + str(size_of_file) + "\n\n"
+            self.logger.info(header_content_type + " " + header_content_length)
+            connection.send('HTTP/1.0 200 OK\r\n'.encode())
+            connection.send(header_content_type.encode())
+            connection.send(header_content_length.encode())
```
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: This stuff basically the same as the previous section. Share code somehow?

File: solutions/abhishek-kujur/aserver.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            with open(file_to_check, 'rb') as file_to_send:
+                read_data = file_to_send.read(1024)
+                connection.send(read_data)
+                while read_data:
+                    read_data = file_to_send.read(1024)
+                    connection.send(read_data)
+            connection.close()
```
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: This code path for reading the file is different from reading an index.html file above. You should share them.

File: solutions/abhishek-kujur/aserver.py (hunk 3 above)
Comment (K-ratos) [inline solutions/abhishek-kujur/aserver.py]: just for source code readability from browser

File: solutions/abhishek-kujur/aserver.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        if(path == '/'):
+            path = ''
+        file_to_check = '.' + path.replace("%20", " ")
+        self.logger.info(file_to_check)
+        self.logger.info("is a file" + str(os.path.isfile(file_to_check)))
+        if os.path.isdir(file_to_check):
+            directory_list = os.listdir(path=file_to_check)
+            if 'index.html' in directory_list:
+                html = open(file_to_check+'/index.html', 'r')
+                r = html.read()
```
Comment (K-ratos) [inline solutions/abhishek-kujur/aserver.py]: yes ... forgot to replace .. nothing popped into my mind at that time so used 'r'

File: solutions/abhishek-kujur/aserver.py (hunk 3 above)
Comment (kaustubh-karkare) [inline solutions/abhishek-kujur/aserver.py]: That's fine.

**Comment by kaustubh-karkare** [comment]
Inactive for too long.

**Pull request closed without merging**